<h3><span class=" glyphicon glyphicon-tag"/>&nbspExceptionVerifier</h3><kbd>Verifies that exceptions are thrown during the test case execution</kbd><br><br><br><h4 style="margin:0px">Class: org.apache.ace.agent.impl.ContentRangeInputStreamTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we cannot read partial content without a Content-Range header.
 */
@Test(expectedExceptions=IOException.class) public void testReadPartialContentWithoutContentRangeHeaderFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new FailingContentConnection(m_content,Failure.PARTIAL_NO_CONTENT_RANGE));
  ContentRangeInputStream is=null;
  try {
    is=new ContentRangeInputStream(handler,m_testURL);
    is.read();
  }
  finally {
    if (is != null) {
      is.close();
    }
  }
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we cannot read partial content if the content is not available.
 */
@Test(expectedExceptions=IOException.class) public void testReadPartialContentNotFoundFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new FailingContentConnection(m_content,Failure.CONTENT_NOT_FOUND));
  ContentRangeInputStream is=null;
  try {
    is=new ContentRangeInputStream(handler,m_testURL);
    is.read();
  }
  finally {
    if (is != null) {
      is.close();
    }
  }
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we cannot read partial content if the server returns a complete body.
 */
@Test(expectedExceptions=IOException.class) public void testReadPartialContentWithCompleteBodyFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new FailingContentConnection(m_content,Failure.PARTIAL_COMPLETE_BODY));
  ContentRangeInputStream is=null;
  try {
    is=new ContentRangeInputStream(handler,m_testURL);
    is.read(new byte[1024]);
    is.read(new byte[1024]);
  }
  finally {
    if (is != null) {
      is.close();
    }
  }
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we can read non-partial content and return the expected contents.
 */
@Test(expectedExceptions=IOException.class) public void testReadClosedStreamFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new CompleteContentConnection(m_content,true));
  ContentRangeInputStream is=new ContentRangeInputStream(handler,m_testURL);
  is.close();
  is.read();
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we cannot read partial content if the server returns a complete body.
 */
@Test(expectedExceptions=IOException.class) public void testReadPartialContentWithChangingContentLengthFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new FailingContentConnection(m_content,Failure.PARTIAL_CHANGING_CONTENT_LENGTH));
  ContentRangeInputStream is=null;
  try {
    is=new ContentRangeInputStream(handler,m_testURL);
    is.read(new byte[1024]);
    is.read(new byte[1024]);
  }
  finally {
    if (is != null) {
      is.close();
    }
  }
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we cannot read partial content if given a non-byte range value in the Content-Range header.
 */
@Test(expectedExceptions=IOException.class) public void testReadPartialContentWithoutByteRangeValueFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new FailingContentConnection(m_content,Failure.PARTIAL_NON_BYTE_RANGE));
  ContentRangeInputStream is=null;
  try {
    is=new ContentRangeInputStream(handler,m_testURL);
    is.read();
  }
  finally {
    if (is != null) {
      is.close();
    }
  }
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that we cannot read partial content if the server is not available.
 */
@Test(expectedExceptions=RetryAfterException.class) public void testReadPartialContentServerUnavailableFail() throws Exception {
  ConnectionHandler handler=new TestConnectionHandler(new FailingContentConnection(m_content,Failure.SERVER_UNAVAILABLE));
  ContentRangeInputStream is=null;
  try {
    is=new ContentRangeInputStream(handler,m_testURL);
    is.read();
  }
  finally {
    if (is != null) {
      is.close();
    }
  }
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.authentication.impl.AuthenticationServiceImplTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that an exception is thrown if no context is given.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testAuthenticateFailsWithoutContext(){
  new AuthenticationServiceImpl().authenticate();
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that an exception is thrown if a null context is given.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testAuthenticateFailsWithNullContext(){
  new AuthenticationServiceImpl().authenticate((Object[])null);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.authentication.processor.basicauth.BasicHttpAuthenticationProcessorTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that a class cast exception is thrown for invalid context when calling authenticate.
 */
@Test(groups={UNIT},expectedExceptions=ClassCastException.class) public void testAuthenticateThrowsClassCastForInvalidContext(){
  new BasicHttpAuthenticationProcessor().authenticate(m_userAdmin,new Object());
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.username" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyKeyUsername() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"");
  props.put(PROPERTY_KEY_PASSWORD,"foo");
  new BasicHttpAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that canHandle throws an {@link IllegalArgumentException} for an empty context.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCanHandleDoesNotAcceptEmptyArray(){
  new BasicHttpAuthenticationProcessor().canHandle(new Object[0]);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.password" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingKeyPassword() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"foo");
  new BasicHttpAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that canHandle throws an {@link IllegalArgumentException} for a null context.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCanHandleDoesNotAcceptNull(){
  new BasicHttpAuthenticationProcessor().canHandle((Object[])null);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.username" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingKeyUsername() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_PASSWORD,"foo");
  new BasicHttpAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.password" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyKeyPassword() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"foo");
  props.put(PROPERTY_KEY_PASSWORD,"");
  new BasicHttpAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.authentication.processor.clientcert.ClientCertAuthenticationProcessorTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "username match policy" property.
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingMatchPolicy() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_USERNAME_LOOKUPKEY,"foo");
  props.put(PROPERTY_VERIFY_CERT_VALIDITY,"true");
  createAuthorizationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "verify cert validity" property.
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingVerifyCertValidity() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_USERNAME_LOOKUPKEY,"foo");
  props.put(PROPERTY_USERNAME_MATCH_POLICY,"foo");
  createAuthorizationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that canHandle throws an {@link IllegalArgumentException} for an empty context.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCanHandleDoesNotAcceptEmptyArray(){
  createAuthorizationProcessor().canHandle(new Object[0]);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that a class cast exception is thrown for invalid context when calling authenticate.
 */
@Test(groups={UNIT},expectedExceptions=ClassCastException.class) public void testAuthenticateThrowsClassCastForInvalidContext(){
  createAuthorizationProcessor().authenticate(m_userAdmin,new Object());
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "username match policy" property.
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyMatchPolicy() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_USERNAME_LOOKUPKEY,"foo");
  props.put(PROPERTY_USERNAME_MATCH_POLICY,"");
  props.put(PROPERTY_VERIFY_CERT_VALIDITY,"true");
  createAuthorizationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "username lookup key" property.
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyLookupKey() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_USERNAME_LOOKUPKEY,"");
  props.put(PROPERTY_USERNAME_MATCH_POLICY,"foo");
  props.put(PROPERTY_VERIFY_CERT_VALIDITY,"true");
  createAuthorizationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "verify cert validity" property.
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyVerifyCertValidity() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_USERNAME_LOOKUPKEY,"foo");
  props.put(PROPERTY_USERNAME_MATCH_POLICY,"bar");
  props.put(PROPERTY_VERIFY_CERT_VALIDITY,"");
  createAuthorizationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "user name lookup key" property.
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingUsernameLookupKey() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_USERNAME_MATCH_POLICY,"foo");
  props.put(PROPERTY_VERIFY_CERT_VALIDITY,"true");
  createAuthorizationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that canHandle throws an {@link IllegalArgumentException} for a null context.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCanHandleDoesNotAcceptNull(){
  createAuthorizationProcessor().canHandle((Object[])null);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.authentication.processor.password.PasswordAuthenticationProcessorTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that canHandle throws an {@link IllegalArgumentException} for an empty context.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCanHandleDoesNotAcceptEmptyArray(){
  new PasswordAuthenticationProcessor().canHandle(new Object[0]);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that canHandle throws an {@link IllegalArgumentException} for a null context.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCanHandleDoesNotAcceptNull(){
  new PasswordAuthenticationProcessor().canHandle((Object[])null);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "password.hashtype" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingPasswordHashType() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"foo");
  props.put(PROPERTY_KEY_PASSWORD,"foo");
  new PasswordAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that a class cast exception is thrown for invalid context when calling authenticate.
 */
@Test(groups={UNIT},expectedExceptions=ClassCastException.class) public void testAuthenticateThrowsClassCastForInvalidContext(){
  new PasswordAuthenticationProcessor().authenticate(m_userAdmin,new Object(),"foo");
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.username" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyKeyUsername() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"");
  props.put(PROPERTY_KEY_PASSWORD,"foo");
  props.put(PROPERTY_PASSWORD_HASHMETHOD,"none");
  new PasswordAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.username" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingKeyUsername() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_PASSWORD,"foo");
  props.put(PROPERTY_PASSWORD_HASHMETHOD,"none");
  new PasswordAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.password" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyKeyPassword() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"foo");
  props.put(PROPERTY_KEY_PASSWORD,"");
  props.put(PROPERTY_PASSWORD_HASHMETHOD,"none");
  new PasswordAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "key.password" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptMissingKeyPassword() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"foo");
  props.put(PROPERTY_PASSWORD_HASHMETHOD,"none");
  new PasswordAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Tests that updated throws an exception for missing "password.hashtype" property. 
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void testUpdatedDoesNotAcceptEmptyPasswordHashType() throws ConfigurationException {
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(PROPERTY_KEY_USERNAME,"foo");
  props.put(PROPERTY_KEY_PASSWORD,"bar");
  props.put(PROPERTY_PASSWORD_HASHMETHOD,"");
  new PasswordAuthenticationProcessor().updated(props);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.builder.DeploymentPackageBuilderTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={Exception.class}) public void testResourceWithoutProcessorDeploymentPackage() throws Exception {
  File tempFile=File.createTempFile("output-",".jar");
  FileOutputStream output=new FileOutputStream(tempFile);
  String name="test";
  String version="1.0.0";
  String pid="my.processor";
  File tempArtifactFile=File.createTempFile("artifact-",".jar");
  DeploymentPackageBuilder.createDeploymentPackage(name,version).addArtifact(createArtifact(pid,tempArtifactFile),pid).generate(output);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.client.repository.helper.base.VelocityArtifactPreprocessorTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test case for {@link VelocityArtifactPreprocessor#preprocess(String,PropertyResolver,String,String,java.net.URL)}
 */
@Test(groups={UNIT},expectedExceptions={IOException.class}) public void testPreprocessNonExistingTemplateOk() throws Exception {
  String url="file:///path/to/nowhere-" + System.currentTimeMillis();
  createProcessor().preprocess(url,m_resolver,TARGET,VERSION1,m_obrUrl);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.client.repository.impl.RepositoryAdminLoginContextImplTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link RepositoryAdminLoginContextImpl#addDescriptor(RepositorySetDescriptor)}.
 */
@Test(groups={TestUtils.UNIT},expectedExceptions={IllegalArgumentException.class}) public void testAddDuplicateObjectRepositoryFail() throws Exception {
  RepositoryAdminLoginContextImpl context=new RepositoryAdminLoginContextImpl(Mockito.mock(User.class),"12345");
  context.addDescriptor(new RepositorySetDescriptor(m_location,CUSTOMER,NAME_SHOP,true,FeatureRepository.class));
  context.addDescriptor(new RepositorySetDescriptor(m_location,CUSTOMER,NAME_DEPLOYMENT,true,FeatureRepository.class));
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link RepositoryAdminLoginContextImpl#addDescriptor(RepositorySetDescriptor)}.
 */
@Test(groups={TestUtils.UNIT},expectedExceptions={IllegalArgumentException.class}) public void testDuplicateRepositoryNameFail() throws Exception {
  RepositoryAdminLoginContextImpl context=new RepositoryAdminLoginContextImpl(Mockito.mock(User.class),"12345");
  context.addDescriptor(new RepositorySetDescriptor(m_location,CUSTOMER,NAME_SHOP,true,ArtifactRepository.class));
  context.addDescriptor(new RepositorySetDescriptor(m_location,CUSTOMER,NAME_SHOP,true,FeatureRepository.class));
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.connectionfactory.impl.ConnectionFactoryImplTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.ConnectionFactoryImpl#createConnection(java.net.URL,org.osgi.service.useradmin.User)}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCreateConnectionNullUserFail() throws Exception {
  new ConnectionFactoryImpl().createConnection(new URL("file:///tmp/foo"),null);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.ConnectionFactoryImpl#createConnection(java.net.URL)}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testCreateConnectionNullUrlFail() throws Exception {
  new ConnectionFactoryImpl().createConnection(null);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.connectionfactory.impl.UrlCredentialsFactoryTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testGetCredentialsWithDictionaryInvalidAuthTypeFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"nonsense");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryBasicTypeMissingUserNameFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"basic");
  props.put(UrlCredentialsFactory.KEY_AUTH_USER_PASSWORD,"bar");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testGetCredentialsWithNullDictionaryFail(){
  UrlCredentialsFactory.getCredentials(null);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryClientCertTypeMissingKeystorePasswordFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"client_cert");
  props.put(UrlCredentialsFactory.KEY_AUTH_TRUSTSTORE_FILE,"bar");
  props.put(UrlCredentialsFactory.KEY_AUTH_TRUSTSTORE_PASS,"qux");
  props.put(UrlCredentialsFactory.KEY_AUTH_KEYSTORE_FILE,"foo");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary,java.lang.String)}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testGetCredentialsWithNullPrefixFail(){
  UrlCredentialsFactory.getCredentials(new Hashtable<String,Object>(),null);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryClientCertTypeMissingTruststorePasswordFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"client_cert");
  props.put(UrlCredentialsFactory.KEY_AUTH_KEYSTORE_FILE,"bar");
  props.put(UrlCredentialsFactory.KEY_AUTH_KEYSTORE_PASS,"qux");
  props.put(UrlCredentialsFactory.KEY_AUTH_TRUSTSTORE_FILE,"foo");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryClientCertTypeMissingKeystoreFileFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"client_cert");
  props.put(UrlCredentialsFactory.KEY_AUTH_TRUSTSTORE_FILE,"bar");
  props.put(UrlCredentialsFactory.KEY_AUTH_TRUSTSTORE_PASS,"qux");
  props.put(UrlCredentialsFactory.KEY_AUTH_KEYSTORE_PASS,"foo");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryClientCertTypeMissingTruststoreFileFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"client_cert");
  props.put(UrlCredentialsFactory.KEY_AUTH_KEYSTORE_FILE,"bar");
  props.put(UrlCredentialsFactory.KEY_AUTH_KEYSTORE_PASS,"qux");
  props.put(UrlCredentialsFactory.KEY_AUTH_TRUSTSTORE_PASS,"foo");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryBasicTypeMissingPasswordFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_BASE_URL,AUTH_BASE_URL);
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"basic");
  props.put(UrlCredentialsFactory.KEY_AUTH_USER_NAME,"bar");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentialsFactory#getCredentials(java.util.Dictionary)}.
 */
@Test(groups={UNIT},expectedExceptions=MissingValueException.class) public void testGetCredentialsWithDictionaryMissingBaseUrlFail(){
  Dictionary<String,Object> props=new Hashtable<>();
  props.put(UrlCredentialsFactory.KEY_AUTH_TYPE,"none");
  UrlCredentialsFactory.getCredentials(props);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.connectionfactory.impl.UrlCredentialsTest </h4><pre class="type-5 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentials#UrlCredentials(org.apache.ace.connectionfactory.impl.UrlCredentials.AuthType,java.net.URL,java.lang.Object[])}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testUrlCredentialsNullTypeFail() throws Exception {
  new UrlCredentials(null,new URL("http://localhost:8080/"));
}

</code></pre>

<br>
<pre class="type-5 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
"></span><br>
/** 
 * Test method for {@link org.apache.ace.connectionfactory.impl.UrlCredentials#UrlCredentials(java.net.URL)}.
 */
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void testUrlCredentialsNullURLFail() throws Exception {
  new UrlCredentials(null);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.discovery.property.SimpleDiscoveryTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test if setting an invalid configuration is handled correctly.
 * @throws ConfigurationException
 */
@Test(groups={UNIT},expectedExceptions=ConfigurationException.class) public void simpleDiscoveryInvalidConfiguration() throws ConfigurationException {
  Dictionary<String,String> properties=new Hashtable<>();
  properties.put(SERVERURL_KEY,INVALID_URL);
  m_discovery.updated(properties);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.log.LogDescriptorTest </h4><pre class="type-5 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
"></span><br>
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void deserializeInvalidDescriptor() throws Exception {
  new Descriptor("invalidStringRepresentation");
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.log.target.store.impl.GatewayLogStoreTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class}) public void testExceptionHandling() throws IOException {
  m_logStore.handleException(m_logStore.getLog(4711),new IOException("test"));
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.obr.storage.file.BundleFileStoreTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class},expectedExceptionsMessageRegExp="Not a valid bundle and no filename found.*") public void putArtifactFail1() throws Exception {
  File bundle=createTmpResource(null,null);
  m_bundleStore.put(new FileInputStream(bundle),null,false);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class},expectedExceptionsMessageRegExp="Not a valid bundle and no filename found.*") public void putArtifactFail2() throws Exception {
  File bundle=createTmpResource(null,null);
  m_bundleStore.put(new FileInputStream(bundle),"",false);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class},expectedExceptionsMessageRegExp="Not a valid bundle and no filename found.*") public void putBundleFail() throws Exception {
  File bundle=createTmpResource(null,"1.0.0");
  String filePath=m_bundleStore.put(new FileInputStream(bundle),null,false);
  assert filePath.equals("foo/bar/foo.bar-1.0.0.jar") : "Path should be 'foo/bar/foo.bar-1.0.0.jar', was " + filePath;
  File file=new File(m_directory,filePath);
  assert file.exists();
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.range.SortedRangeSetTest </h4><pre class="type-5 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
"></span><br>
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public void invalidRange(){
  new SortedRangeSet("8-5");
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.repository.impl.RepositoryImplTest </h4><pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testGetVersionZeroForEmptyRepositoryFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.get(0);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class}) public void testCommitNonExistingVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  InputStream data=new ByteArrayInputStream("abc".getBytes());
  repo.commit(data,0);
  data=new ByteArrayInputStream("def".getBytes());
  repo.commit(data,2);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testCheckoutVersionZeroOnEmptyRepositoryFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.checkout(0);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testGetNegativeVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.get(-1);
}

</code></pre>

<br>
<pre class="type-7 type-2 type-10 type-1 type-5 type-9 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies values of objects/variables related to API calls (Java or TPL) ">APIUtilityVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies values of objects/variables related to AUT calls">InternalCallVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies boolean conditions">BooleanVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies whether objects/variable are equal to an expected value ">EqualityVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Contains more than 2 JUnit-based stereotypes">HybridVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies values of objects/variables related to API calls (Java or TPL) 
- Verifies values of objects/variables related to AUT calls
- Verifies boolean conditions
- Verifies whether objects/variable are equal to an expected value 
- Verifies that exceptions are thrown during the test case execution
- Contains more than 2 JUnit-based stereotypes
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalStateException.class}) public void testUpdatedConfigurationOk() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  File file=new File(m_baseDir,"newLocation" + File.separator + "1");
  repo.updated(false,Long.MAX_VALUE);
  assertFalse(repo.commit(new ByteArrayInputStream("abc".getBytes()),0),"Committing should not be allowed on slave repositories.");
  assertTrue(repo.put(new ByteArrayInputStream("abc".getBytes()),1),"'put'ting a replica should be allowed on slave repositories.");
  String readLine=getContentAsString(file);
  assertEquals(readLine,"abc","File " + file.getAbsolutePath() + " should have contained 'abc'.");
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testPutVersionZeroFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.put(new ByteArrayInputStream("abc".getBytes()),0);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testPutNegativeVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.put(new ByteArrayInputStream("abc".getBytes()),-1);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class}) public void testCommitIncorrectVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  InputStream data=new ByteArrayInputStream("abc".getBytes());
  repo.commit(data,1);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testCheckoutNegativeVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.checkout(-1);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IOException.class}) public void testCommitExistingVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  InputStream data=new ByteArrayInputStream("abc".getBytes());
  repo.commit(data,0);
  data=new ByteArrayInputStream("def".getBytes());
  repo.commit(data,0);
}

</code></pre>

<br>
<pre class="type-5 type-4 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(groups={UNIT},expectedExceptions={IllegalArgumentException.class}) public void testCommitNegativeVersionFail() throws Exception {
  RepositoryImpl repo=new RepositoryImpl(new File(m_baseDir,"data"),new File(m_baseDir,"tmp"),true);
  repo.commit(new ByteArrayInputStream("abc".getBytes()),-1);
}

</code></pre>

<br>
<h4 style="margin:0px">Class: org.apache.ace.scheduler.SchedulerTest </h4><pre class="type-5 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
"></span><br>
@Test(groups={UNIT},expectedExceptions=IllegalArgumentException.class) public synchronized void testIllegalCreation(){
  new SchedulerTask(null);
}

</code></pre>

<br>
<script>$(document).ready(function() {
  $('pre code').each(function(i, block) {
    hljs.highlightBlock(block);
  });
});</script>
